lib/tree.rb

```diff
@@ -30,6 +30,14 @@ def contents_array
       @contents_ary ||= jtree_entries
     end
     
+    def recursive_count(limit = nil)
+      jtree_entries({recursive: true, limit: limit}).size
```

Comment:
Caching the different limits is very cheap memory-wise (a hash with integer keys and values), while the computation can be quite expensive. On the other hand, it seems likely that users would store the result in a variable themselves, so maybe it's too much. Dunno. :)